cassovary-core/src/main/scala/com/twitter/cassovary/util/io/AdjacencyListGraphReader.scala

```diff
@@ -73,26 +73,29 @@ class AdjacencyListGraphReader[T] (
    * @param filename Name of file to read from
    */
   private class OneShardReader(filename: String, nodeNumberer: NodeNumberer[T])
-    extends Iterator[NodeIdEdgesMaxId] {
+    extends Iterable[NodeIdEdgesMaxId] {
 
-    private val outEdgePattern = ("""^(\w+)""" + separator + """(\d+)""").r
-    var lastLineParsed = 0
-    private val lines = Source.fromFile(filename).getLines()
-      .map{x => {lastLineParsed += 1; x}}
-    private val holder = NodeIdEdgesMaxId(-1, null, -1)
+    override def iterator = new Iterator[NodeIdEdgesMaxId] {
 
-    override def hasNext: Boolean = lines.hasNext
+      private val outEdgePattern = ("""^(\w+)""" + separator + """(\d+)""").r
```

Comment:
you can keep immutables (such as definition of patterns) outside the iterator if you want